vmngclient/api/tenant_migration_api.py

```diff
@@ -0,0 +1,193 @@
+            wan_edge_forecast (int): Forecasted number of WAN Edges for given tenant.
+        Returns:
+            Task: object representing initiated export process
+        """
+        tenant = self._create_tenant(**kwargs)
+        process_id = self.session.primitives.tenant_migration.export_tenant_data(tenant).process_id
+        return Task(self.session, process_id)
+
+    def download(self, download_path: Path, remote_filename: str = "default.tar.gz"):
+        """Download exported single-tenant deployment and configuration data from a Cisco vManage instance
+        to a local file system. Should be executed on migrated single-tenant system.
+
+        Args:
+            download_path (Path): full download path containing a filename eg.: Path("/home/user/tenant-export.tar.gz")
+            remote_filename (str): path to exported tenant migration file on vManage
+        """
+        tenant_data = self.session.primitives.tenant_migration.download_tenant_data(remote_filename)
+        with open(download_path, "wb") as file:
+            file.write(tenant_data)
+
+    def import_tenant(self, import_file: Path) -> ImportTask:
+        """Imports the single-tenant deployment and configuration data into multi-tenant vManage instance.
+        Should be executed on target multi-tenant system.
+
+        Args:
+            import_file (Path): full path to previously exported single-tenant data file
+
+        Returns:
+            ImportTask: object representing initiated import process
+        """
+        import_info = self.session.primitives.tenant_migration.import_tenant_data(open(import_file, "rb"))
+        return ImportTask(self.session, import_info)
+
+    def store_token(self, migration_id: str, download_path: Path):
+        """Stores migration token as text file for given migration identifier.
+        Should be executed on target multi-tenant system.
+
+        Args:
+            migration_id (str): migration identifier (it is obtained after import tenant task is finished)
+            download_path (Path): full download path containing a filename eg.: Path("/home/user/import-token.txt")
+        """
+        params = MigrationTokenQueryParams(migrationId=migration_id)
+        token = self.session.primitives.tenant_migration.get_migration_token(params)
+        with open(download_path, "w") as file:
+            file.write(token)
+
+    def migrate_network(self, token_file: Path) -> Task:
+        """Starts migration procedure on single-tenant system.
+
+        Args:
+            token_file (Path): full path to previously stored text file with import token.
+
+        Returns:
+            Task: object representing initiated migration process
+        """
+        with open(token_file, "r") as file:
+            token = file.read()
+        process_id = self.session.primitives.tenant_migration.migrate_network(token).process_id
+        return Task(self.session, process_id)
+
+
+@overload
+def st_to_mt(
```

Comment:
Do we plan in the future to put `workflows` like that in some dedicated space?

Reply:
Not sure...
This procedure can be executed using merely tenant migration APIs so it fits here.
Other workflows may mix different apis or even use other interfaces than vManage REST.